core/arch/arm/kernel/otp_stubs.c

```diff
@@ -23,13 +23,15 @@ __weak TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
 	return TEE_SUCCESS;
 }
 
+#if !defined(CFG_NXP_SE05X_HUK_DRV)
```

Comment:
unfortunately that is not what happens with weak definitions and libraries. see the commit that works around the issue.

Reply:
I think here, SE05x driver should implement `tee_otp_get_die_id()` to override this weak function and call the target library handler.

Reply:
but all the se050 code is just a library, there is no associated driver....do you think we should have a proper se050x driver for things like this?

Reply:
I think SE05x could register a so-called crypto driver (see `CFG_CRYPTO_DRIVER`) to be called for the crypto operation is supports.

Reply:
@etienne-lms, interesting approach....is that the plan for libmbedtls?

Reply:
so, instead of using macros to configure the functionality that the user needs from tomcrypt, the request is that we write specific drivers to override those operations. 
if you could confirm before I put the time doing this it would be much appreciated. thanks.

Reply:
If implementing specific crypto operations, I think introducing a crpytodrv for what SE05x is expected to support seems appropriate. See how caam handles some RSA operations in [.../crypto/.../caam_rsa.c](https://github.com/OP-TEE/optee_os/blob/3.10.0/core/drivers/crypto/caam/acipher/caam_rsa.c#L1669).

As for confirming it is the way to go, I would rather ask @jforissier and @jenswi-linaro advices.